File: app/src/main/java/com/nawin/androidmvparchitecture/auth/login/LoginActivity.java
package com.nawin.androidmvparchitecture.auth.login;

import android.app.ProgressDialog;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.widget.Toast;

import com.nawin.androidmvparchitecture.BaseActivity;
import com.nawin.androidmvparchitecture.R;
import com.nawin.androidmvparchitecture.databinding.ActivityLoginBinding;
import com.nawin.androidmvparchitecture.taggedquestion.TaggedQuestionsActivity;
import com.nawin.androidmvparchitecture.utils.Commons;

import javax.inject.Inject;

import static android.text.TextUtils.isEmpty;

public class LoginActivity extends BaseActivity implements LoginContract.View {
    @Inject
    LoginContract.Presenter presenter;
    private ProgressDialog progressDialog;
    private ActivityLoginBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login);
//        new LoginPresenter(component, this);

        binding.btnLogin.setOnClickListener(v -> {
            String username = binding.etUsername.getText().toString();
            if (isEmpty(username)) {
                binding.etUsername.setError(getString(R.string.empty_field_msg));
                return;
            }
            String password = binding.etPassword.getText().toString();
            if (isEmpty(password)) {
                binding.etPassword.setError(getString(R.string.empty_field_msg));
                return;
            }
            presenter.onLogin(username, password);
        });
    }

    @Override
    protected void onPause() {
        presenter.stop();
        super.onPause();
    }

    @Override
    public void showLoginProgress() {
        progressDialog = Commons.showLoadingDialog(this, getString(R.string.verifying_login));
    }

    @Override
    public void showLoginSuccess() {
        dismissDialog();
        TaggedQuestionsActivity.start(this);
        finish();
    }

    @Override
    public void showLoginError(String message) {
        dismissDialog();
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();

    }

    @Override
    public void showNetworkNotAvailableError() {
        dismissDialog();
        Toast.makeText(this, getString(R.string.network_not_available_error), Toast.LENGTH_SHORT).show();
    }

//    @Override
//    public void setPresenter(LoginContract.Presenter presenter) {
//        this.presenter = presenter;
//    }

    private void dismissDialog() {
        if (progressDialog != null && progressDialog.isShowing())
            progressDialog.dismiss();
    }

}

File: app/src/main/java/com/nawin/androidmvparchitecture/auth/login/LoginActivity.kt
package com.nawin.androidmvparchitecture.auth.login

/**
 * Created on 12/13/17.
 */
import android.app.ProgressDialog
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.widget.Toast

import com.nawin.androidmvparchitecture.BaseActivity
import com.nawin.androidmvparchitecture.R
import com.nawin.androidmvparchitecture.databinding.ActivityLoginBinding
import com.nawin.androidmvparchitecture.taggedquestion.TaggedQuestionsActivity

import android.text.TextUtils.isEmpty
import com.nawin.androidmvparchitecture.utils.showLoadingDialog

class LoginActivity : BaseActivity(), LoginContract.View {
    private lateinit var presenter: LoginContract.Presenter
    private var progressDialog: ProgressDialog? = null
    private lateinit var binding: ActivityLoginBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_login)
        LoginPresenter(component, this)

        binding.btnLogin.setOnClickListener {
            val username = binding.etUsername.text.toString()
            val password = binding.etPassword.text.toString()
            if (isEmpty(username) && isEmpty(password)) {
                binding.etUsername.error = getString(R.string.empty_field_msg)
                binding.etPassword.error = getString(R.string.empty_field_msg)
            } else if (isEmpty(username)) {
                binding.etUsername.error = getString(R.string.empty_field_msg)
            } else if (isEmpty(password)) {
                binding.etPassword.error = getString(R.string.empty_field_msg)
            } else {
                presenter.onLogin(username, password)

            }

        }
    }

    override fun onPause() {
        presenter.stop()
        super.onPause()
    }

    override fun showLoginProgress() {
        progressDialog = showLoadingDialog(this, getString(R.string.verifying_login))
    }

    override fun showLoginSuccess() {
        dismissDialog()
        TaggedQuestionsActivity.start(this)
        finish()
    }

    override fun showLoginError(message: String) {
        dismissDialog()
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
    }

    override fun showNetworkNotAvailableError() {
        dismissDialog()
        Toast.makeText(this, getString(R.string.network_not_available_error), Toast.LENGTH_SHORT).show()
    }

    override fun setPresenter(presenter: LoginContract.Presenter) {
        this.presenter = presenter
    }

    private fun dismissDialog() {
            progressDialog!!.dismiss()
    }

}
